.github/workflows/ci.yml

```diff
@@ -1,39 +1,37 @@
 name: CI
-on:
-  push:
-    branches: [master]
-  pull_request:
-    branches: [master]
+on: [push, pull_request]
 
 jobs:
   test:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        mongodb: ["4.4"]
-        ruby: ["2.5", "2.6", "2.7", "3.0"]
-        gemfile:
-          - mongoid-3.1
-          - mongoid-4.0
-          - mongoid-5.0
-          - mongoid-6.0
-          - mongoid-7.0
-          - mongoid-8.0
-
+        entry:
+          - { ruby: "2.6", mongodb: '4.4', gemfile: 'mongoid-4.0', bundler: '2.0.0.pre.3' }
+          - { ruby: "2.7", mongodb: '4.4', gemfile: 'mongoid-5.0', bundler: '2.0.0.pre.3' }
+          - { ruby: "2.7", mongodb: '4.4', gemfile: 'mongoid-6.0' }
+          - { ruby: "2.7", mongodb: '4.4', gemfile: 'mongoid-7.0' }
+          - { ruby: "3.1", mongodb: '4.4', gemfile: 'mongoid-8.0' }
```

Comment:
[CONFUSED] I don't understand this refactor of the matrix.  I'd rather see a traditional build matrix and possible `include` or `exclude` keys for edge cases. 

i.e. This `entry` array doesn't make sense to me. What am I misunderstanding?

Reply:
1. This is a 6x4 matrix and we should also be testing multiple MongoDB versions, so this will be 6x4x3 = 72 builds where about 75% are completely redundant. Even here, I started with a regular matrix and ended up with a long list of excludes, this avoids doing that.
2. For these older Rubies we need a different version of bundler (2.0.0.pre.3). This is yet another parameter then, matrix explodes.